cmd/remediation-aggregator/main.go

```diff
@@ -0,0 +1,426 @@
+			return err
+		}
+	}
+
+	return nil
+}
+
+func readContent(filename string) (string, error) {
+	// gosec complains that the file is passed through an evironment variable. But
+	// this is not a security issue because none of the files are user-provided
+	cleanFileName := filepath.Clean(filename)
+	// #nosec G304
+	file, err := os.Open(cleanFileName)
+	if err != nil {
+		return "", err
+	}
+	defer file.Close()
+
+	b, err := ioutil.ReadAll(file)
+	return string(b), err
+}
+
+func aggregator(cmd *cobra.Command, args []string) {
+	var scanRemediations []*complianceoperatorv1alpha1.ComplianceRemediation
+
+	aggregatorConf := parseConfig(cmd)
+
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		// fallback to kubeconfig
+		fmt.Println("Running outside cluster")
+		kubeconfig := filepath.Join("~", ".kube", "config")
+		if envvar := os.Getenv("KUBECONFIG"); len(envvar) > 0 {
+			kubeconfig = envvar
+		}
+		fmt.Printf("Using KUBECONFIG %s\n", kubeconfig)
+		config, err = clientcmd.BuildConfigFromFlags("", kubeconfig)
+		if err != nil {
+			fmt.Printf("The kubeconfig cannot be loaded: %v\n", err)
+			os.Exit(1)
+		}
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		fmt.Printf("Cannot create clientset: %v\n", err)
+		os.Exit(1)
+	}
+
+	crclient, err := createCrClient(config)
+	if err != nil {
+		fmt.Printf("Cannot create client for our types: %v\n", err)
+		os.Exit(1)
+	}
+
+	var scan = &complianceoperatorv1alpha1.ComplianceScan{}
+	err = crclient.client.Get(context.TODO(), types.NamespacedName{
+		Namespace: aggregatorConf.Namespace,
+		Name:      aggregatorConf.ScanName,
+	}, scan)
+	if err != nil {
+		fmt.Printf("Cannot retrieve the scan instance: %v\n", err)
+		os.Exit(1)
+	}
+
+	if scan.Status.Result == complianceoperatorv1alpha1.ResultError {
+		fmt.Println("Not gathering results from a scan that resulted in an error")
+		os.Exit(0)
+	}
+
+	content, err := readContent(aggregatorConf.Content)
+	if err != nil {
+		fmt.Printf("Cannot read the content: %v\n", err)
+		os.Exit(1)
+	}
+
+	// Find all the configmaps for a scan
+	configMaps, err := getScanConfigMaps(clientset, aggregatorConf.ScanName, aggregatorConf.Namespace)
+	if err != nil {
+		fmt.Printf("getScanConfigMaps failed: %v\n", err)
+		os.Exit(1)
+	}
+
+	// For each configmap, create a list of remediations
+	for _, cm := range configMaps {
+		fmt.Printf("processing CM: %s\n", cm.Name)
+		cmRemediations, err := parseResultRemediations(crclient.scheme, aggregatorConf.ScanName, aggregatorConf.Namespace, content, &cm)
+		if err != nil {
+			fmt.Printf("Cannot parse CM %s into remediations, err: %v\n", cm.Name, err)
+		} else if cmRemediations == nil {
+			fmt.Println("Either no remediations found in result or result already processed")
+			continue
+		}
+		fmt.Printf("CM %s contained %d remediations\n", cm.Name, len(cmRemediations))
+
+		// Annotate the configmap, we want to avoid parsing it next time the reconcile
+		// loop hits
```

Comment:
We wouldn't need this comment with a better function name. e.g. annotateParsedConfigMap

Reply:
OK, fixed